src/ore/src/tracing.rs

```diff
@@ -463,21 +457,51 @@ where
     Ok((handle, guard))
 }
 
-/// Returns the level of a specific target from a [`Targets`].
-pub fn target_level(targets: &Targets, target: &str) -> Level {
-    if targets.would_enable(target, &Level::TRACE) {
-        Level::TRACE
-    } else if targets.would_enable(target, &Level::DEBUG) {
-        Level::DEBUG
-    } else if targets.would_enable(target, &Level::INFO) {
-        Level::INFO
-    } else if targets.would_enable(target, &Level::WARN) {
-        Level::WARN
-    } else {
-        Level::ERROR
+struct EmptyCallsite;
+impl Callsite for EmptyCallsite {
+    fn set_interest(&self, _: Interest) {
+        unreachable!("EmptyCallsite should never be accessed")
+    }
+
+    fn metadata(&self) -> &Metadata<'_> {
+        unreachable!("EmptyCallsite should never be accessed")
     }
 }
 
+/// Returns the [`Level`] of a crate from an [`EnvFilter`] by performing an
+/// exact match between `crate` and the original `EnvFilter` directive.
+pub fn crate_level(filter: &EnvFilter, crate_name: &'static str) -> Level {
+    // TODO: implement `would_enable` on `EnvFilter` or equivalent
+    // to avoid having to manually parse out the directives. This
+    // would also significantly broaden the lookups the fn is able
+    // to do (modules, spans, fields, etc).
+
+    let mut default_level = Level::ERROR;
+    // EnvFilter roundtrips through its Display fmt, so it
+    // is safe to split out its individual directives here
+    for directive in format!("{}", filter).split(',') {
+        match directive.split('=').collect::<Vec<_>>().as_slice() {
+            [target, "=", level] => {
```

Comment:
This block seems interesting and uncovered.

Reply:
in retrospect, this function is very easy to unit test! lemme go add those in